### PR TITLE
make tests run and fix all tests

### PR DESCRIPTION
**Pull request by glebtv**
tested on mongoid 6.0, 6.1, 6.2

### CODE REVIEW

**Comment by glebtv** [comment]
travis fails on jruby - seems to be failing to install cancancan 2.0

**Comment by coorasse** [comment]
Seems like it fails to install bundler. To me you can drop the support for jruby for this gem. What do you think?

File: lib/cancancan/model_adapters/mongoid_adapter.rb [hunk 1]
```diff
@@ -19,7 +19,12 @@ def self.override_conditions_hash_matching?(subject, conditions)
       def self.matches_conditions_hash?(subject, conditions)
         # To avoid hitting the db, retrieve the raw Mongo selector from
         # the Mongoid Criteria and use Mongoid::Matchers#matches?
-        subject.matches?(subject.class.where(conditions).selector)
+        q = subject.class.where(conditions).selector
+        if subject.respond_to?(:_matches?)
+          subject._matches?(q)
+        else
+          subject.matches?(q)
```
Comment (coorasse) [inline lib/cancancan/model_adapters/mongoid_adapter.rb:26]: is this required only for Mongoid 2? I'd drop support for it

**Review by coorasse — APPROVED** [review]
Very nice job! What if we drop support for mongoid 2 and jruby so all the tests are green and we can remove some very old code?

**Comment by shaunakpp** [comment]
Hello,
@glebtv, I did a pull of your branch and it seems that the `Appraisals` file is not updated to handle the revised gemfiles, could you please update the PR with the file so that I can run these tests locally ?

**Pull request merged**